runtime/src/common/bytes.rs

```diff
@@ -171,37 +175,90 @@ macro_rules! impl_bytes {
                         &self,
                         formatter: &mut ::std::fmt::Formatter,
                     ) -> ::std::fmt::Result {
-                        formatter.write_str("bytes or sequence of u8")
+                        formatter.write_str("bytes or string expected")
                     }
 
-                    fn visit_seq<A>(self, mut seq: A) -> Result<$name, A::Error>
+                    fn visit_str<E>(self, data: &str) -> Result<$name, E>
                     where
-                        A: ::serde::de::SeqAccess<'de>,
+                        E: ::serde::de::Error,
                     {
                         let mut array = [0; $size];
-                        for i in 0..$size {
-                            array[i] = seq
-                                .next_element()?
-                                .ok_or_else(|| ::serde::de::Error::invalid_length(i, &self))?;
+                        let bytes = base64::decode(data).unwrap();
```

Comment:
This should not unwrap, otherwise a serialization error will cause the program to panic instead of returning a serialization error. Handle the error and return a proper error type instead.